codegen/src/main/kotlin/apifi/codegen/ApiBuilder.kt

```diff
@@ -6,13 +6,16 @@ import com.squareup.kotlinpoet.*
 
 class ApiBuilder(private val apiMethodBuilder: ApiMethodBuilder = ApiMethodBuilder()) {
 
-    fun build(name: String, paths: List<Path>, basePackageName: String, modelMapping: Map<String, String>): FileSpec {
+    fun build(name: String, paths: List<Path>, basePackageName: String, modelMapping: Map<String, String>, securityProvider: SecurityProvider): FileSpec {
         val baseName = name.toTitleCase()
         val controllerClassName = "${baseName}Api"
 
-        val controllerInterfaceClass = ControllerInterfaceBuilder.build(paths, baseName)
+        val controllerInterfaceClass = ControllerInterfaceBuilder.build(paths, baseName, securityProvider)
 
-        val allControllerFunSpecs = paths.flatMap { path -> path.operations?.map { op -> apiMethodBuilder.methodFor(path.url, op, modelMapping) }?: emptyList() }
+        val allControllerFunSpecs = paths.flatMap { path ->
+            path.operations?.map { op -> apiMethodBuilder.methodFor(path.url, op, modelMapping, securityProvider) }
```

Comment:
@bhimsenp I was thinking this code might not be maintainable if we add params other than that of path and operation to `apiMethodBuilder.methodFor`. Since `modelMapping` and `securityProvider` are more of global things, can we call them directly in `apiMethodBuilder.methodFor` somehow?

Reply:
Possible for SecurityProvider but not for modelMapping. We can use SecurityProvider as Object (similar to PathBuilder, ModelBuilder etc.). But modelMapping has to be passed as parameter.

Reply:
Okay. Lets keep `modelMapping` as is then.